### ISSUE TITLE
[RFC] directive to vizualize change detection

### ISSUE THREAD

**Issue by tomalaforge** [nartc/ngxtension-platform#85]
Would it make them to provide a directive to test change detection inside your application ? To test or debug CD issues.

```ts
import { Directive, DoCheck, ElementRef, NgZone } from '@angular/core';

@Directive({
  selector: '[cd-flash]',
  standalone: true,
})
export class CDFlashingDirective implements DoCheck {
  constructor(private elementRef: ElementRef, private zone: NgZone) {}

  ngDoCheck(): void {
    this.cdRan();
  }

  public cdRan(): void {
    this.zone.runOutsideAngular(() => {
      this.elementRef.nativeElement.classList.add('!bg-orange-500');
      setTimeout(() => {
        this.elementRef.nativeElement.classList.remove('!bg-orange-500');
      }, 1000);
    });
  }
}
```

You just need to add it on a element to vizualise when CD is made inside that element. 

What do you think ?

**Comment by wanoo21** [comment]
Maybe this is the case to check if [isDevMode()](https://angular.io/api/core/isDevMode), to not have "surprises" in production 😉

**Comment by ducin** [comment]
IMO it sounds cool and useful for debugging/educational purposes.
Is it decided to be done, or do you need any help with that guys? @tomalaforge @nartc ?

**Comment by tomalaforge** [comment]
@ducin Yes I need to take the time to do it. It's already done on my angular challenge project, just need to copy paste it and write some documentation.

**Comment by endlacer** [comment]
When multiple CDs run in a short period you would miss that, cause it is either on or off during 1s.
i am thinking of the component flashing up and then slowly fading away during this 1 second. just like space-sniffer is doing: eg: https://www.youtube.com/watch?v=lUnsyAIZO2Q
